test/Dialect/HW/specialize.mlir

```diff
@@ -0,0 +1,120 @@
+// RUN: circt-opt %s --split-input-file --hw-specialize | FileCheck %s
+
+// Test two different ways of instantiating a generic module.
+
+module {
+
+// CHECK-LABEL:  hw.module @addToFirst_N_4_X_8(%vec: !hw.array<4xi8>, %a: i8) -> (out: i8) {
+// CHECK:          %c0_i64 = arith.constant 0 : i64
+// CHECK:          %0 = comb.extract %c0_i64 from 0 : (i64) -> i2
+// CHECK:          %1 = hw.array_get %vec[%0] : !hw.array<4xi8>
+// CHECK:          %2 = comb.add %1, %a : i8
```

Comment:
nit: these tests look good, but please use the `%[[A:+]]` form to make them agnostic to SSA value names. You also might be able to elide some other information if it's not totally crucial, like the type of the hw.output op for instance.